src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -140,15 +140,10 @@ private unsafe Socket(SafeSocketHandle handle, bool loadPropertiesFromHandle)
             try
             {
                 // Get properties like address family and blocking mode from the OS.
-                LoadSocketTypeFromHandle(handle, out _addressFamily, out _socketType, out _protocolType, out _willBlockInternal);
-
-                // Determine whether the socket is in listening mode.
-                _isListening =
-                    SocketPal.GetSockOpt(_handle, SocketOptionLevel.Socket, SocketOptionName.AcceptConnection, out int isListening) == SocketError.Success &&
-                    isListening != 0;
+                LoadSocketTypeFromHandle(handle, out _addressFamily, out _socketType, out _protocolType, out _willBlockInternal, out _isListening);
 
                 // Try to get the address of the socket.
-                Span<byte> buffer = stackalloc byte[512]; // arbitrary high limit that should suffice for almost all scenarios
+                Span<byte> buffer = stackalloc byte[SocketPal.MaximumAddressSize];
```

Comment:
It'd be good to assert that MaximumAddressSize is something reasonable to stackalloc, e.g.
```C#
Debug.Assert(SocketPal.MaximumAddressSize <= 512);
```

Reply:
added